### PR TITLE
golangci-lint: add more linters

### PR DESCRIPTION
**Pull request by katexochen**

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -26,25 +26,34 @@ linters:
     - unused
     # Additional linters
     - bodyclose
```
Comment (burgerdev) [inline .golangci.yml:28]:
```suggestion
    # Additional linters
    # keep-sorted start
    - bodyclose
```

File: .golangci.yml [hunk 2]
```diff
@@ -26,25 +26,34 @@ linters:
...
-    - contextcheck
-    - errchkjson
-    - errorlint
-    - gocheckcompilerdirectives
-    - nilerr
-    - predeclared
-    - sloglint
-    - testifylint
     - usestdlibvars
+    - wastedassign
```
Comment (burgerdev) [inline .golangci.yml:56]:
```suggestion
    - wastedassign
    # keep-sorted end
```

**Review by burgerdev — APPROVED** [review]

File: node-installer/node-installer.go [hunk 3]
```diff
@@ -342,10 +342,12 @@ func ensureMapPath(in *map[string]any, path ...string) map[string]any {
...
 	for _, p := range path {
-		if current[p] == nil {
-			current[p] = make(map[string]any)
+		cur, ok := current[p].(map[string]any)
+		if !ok || cur == nil {
+			cur = make(map[string]any)
+			current[p] = cur
 		}
-		current = current[p].(map[string]any)
+		current = cur
```
Comment (katexochen) [inline node-installer/node-installer.go:350]: @burgerdev had to add this non-trivial fix, could you take a look?

Comment (burgerdev) [reply node-installer/node-installer.go:350]: seems good.

**Review by burgerdev — APPROVED** [review]

**Pull request merged**